### PR TITLE
Add new map "Cosmic Catastrophe" by s20

### PR DESCRIPTION
**Pull request by src4026**
## To-do
- [x] Conversion of original PR to four team
- Map Related Bugs:
    - [ ] Fix misaligned blocks from rotation
    - [ ] Check border mixing from each quadrant
- [ ] Update sewers and make map deeper
- [ ] Update team colours
- [ ] Make sky box (if possible in the time frame I have)

### CODE REVIEW

**Comment by LuismiInTheZone** [comment]
Look:

Team blue: 38 mese, 27 diamonds
Team red: 32 mese, 32 diamonds

I like what you did of trolling people at red base putting the ores at the floating base lol

Also there are some parts from the barrier from the botton that are missing, I recomend you doing again the process of putting the indestructible glass

And for now... everything looks ok, also another 2 things:

1. I recomend putting that zking, FlammingRC... helped you at the part of other
2. What does means et al?

**Comment by src4026** [comment]
> Look:
> 
> Team blue: 38 mese, 27 diamonds
> Team red: 32 mese, 32 diamonds
> 
> I like what you did of trolling people at red base putting the ores at the floating base lol
> 
> Also there are some parts from the barrier from the botton that are missing, I recomend you doing again the process of putting the indestructible glass
> 
> And for now... everything looks ok, also another 2 things:
> 
> 1. I recomend putting that zking, FlammingRC... helped you at the part of other
> 2. What does means et al?

Since the ores diamonds on team blue were a bit less, as you mentioned, I have added four more. As for the indestructible glass, I cannot find the area you are saying are missing. Can you please mention the coordinates? I moved the authors to the info section. (I haven't tested how it looks in-game). "Et al" is a way to say that something (here the map) was made by me, (the member I mentioned) and various others who suggested and helped in the objective making of the map.

**Comment by LuismiInTheZone** [comment]
> > Look:
> > Team blue: 38 mese, 27 diamonds
> > Team red: 32 mese, 32 diamonds
> > I like what you did of trolling people at red base putting the ores at the floating base lol
> > Also there are some parts from the barrier from the botton that are missing, I recomend you doing again the process of putting the indestructible glass
> > And for now... everything looks ok, also another 2 things:
> > 
> > 1. I recomend putting that zking, FlammingRC... helped you at the part of other
> > 2. What does means et al?
> 
> Since the ores diamonds on team blue were a bit less, as you mentioned, I have added four more. As for the indestructible glass, I cannot find the area you are saying are missing. Can you please mention the coordinates? I moved the authors to the other info section. (I haven't tested how it looks in-game). "Et al" is a way to say that something (here the map) was made by me, (the members I mentioned) and various others who suggested and helped in the objective making of the map indirectly, not through direct building participation. Et al is used in official papers and documents but I decided to employ it here as a short way to include all those who contributed to the map. (Note: the way I had used "et al" isn't the way it is supposed to be, technically).

It doesn't matter to find the cordinates, just select all the botton layer and add again ind_glass... Also did you fix the mese?

**Comment by src4026** [comment]
Oops

**Comment by MoreKat** [comment]
> "Et al" is a way to say that something (here the map) was made by me, (the members I mentioned) and various others who suggested and helped in the objective making of the map indirectly, not through direct building participation. Et al is used in official papers and documents but I decided to employ it here as a short way to include all those who contributed to the map. (Note: the way I had used "et al" isn't the way it is supposed to be, technically).

Why don't you just use 'and others'? This is not an academic setting and I think it's better to use language that everyone will understand especially considering CTF's playerbase.

I will test the map later btw.

**Comment by src4026** [comment]
> Why don't you just use 'and others'? This is not an academic setting and I think it's better to use language that everyone will understand especially considering CTF's playerbase.
> 
> I will test the map later btw.

In commit [#9e66d90](https://github.com/MT-CTF/maps/pull/286/commits/9e66d90d4d9ab23b52ed6e91b099d6f0607807b3) I had also moved the authors to the `Other Info` field, so it won't be visible upon the beginning information of the map. I want to keep the "et al" in the `Other Info` section for two reasons, (1) I really want to use it :smile: and (2) Since more the 50-70% of players won't rly go in /maps a lot and when they do, I would like them to figure it out by searching it online or whatever may be their method. 

```conf
....
name = Cosmic Catastrophe
phys_gravity = 1
initial_stuff = return {"default:pick_steel","default:cobble 30","default:torch 5","ctf_ranged:pistol_loaded"}
author = s20
hint = Red is the sky, Blue is the water. And beneath is the System
license = CC BY-SA 4.0
others = Made with the assistance of zking, FlamingRCCars, PantsMan et al.
....
```

**Review by MoreKat — COMMENTED** [review]
I'm not sure what to think of this map. First of all you should fix the differences between both teams' treasure (ore) rooms. Blue's ladder is covered with a block and theirs is completely dark unlike red's. I don't think the hint `Red is the sky, Blue is the water. And beneath is the System.` is particularly helpful. Aside from the fact that the sky isn't red (?), I think it is too vague. I can see that you're implying that players should look underground (presumably to find the ores?) but this should be clearer especially since the ores are quite hard to find. Maybe you mean something else with the 'System', not sure. 
The map is also quite similar to the recently added Savanna Ravine map. We have a map with a ravine filled with lava, one filled with water, I don't think we need one filled with kill nodes. I actually think it would be cool to have a map with a ravine you can walk and fight in. Besides, I think the instant kill nodes will cause a lot of frustration for the players. They will die often, also when they try to tunnel to get to the other side. I've noticed that the kill nodes aren't visible when you look straight at them (as you would do when tunneling, see picture). 
screenshot_20230505_163758
In general I think the map as it is now will be hard to capture, except maybe in Nade Fight as you can just launch yourself over the ravine with the yellow grenade. You should also take into account that maps with mese and diamond are generally harder to capture.
Lastly, why did you change the name to '**The** Cosmic Catastrophe'? IMO, just 'Cosmic Catastrophe' sounds much better. 
As to the 'et al' thing, I don't think your reasons make sense but it's up to you so go ahead.

**Comment by src4026** [comment]
I have fixed the unbalanced ores. The torch seems to be left from my building sessions, I will remove that. I think I should better make the ravine with stone and not kill nodes. ~~I am trying capturing to be mid hard.nut it doesn't matter to me if it is hard. :P. ~~ I will also look into making a better hint. Thanks for the feedback :thumbsup:

**Comment by MoreKat** [comment]
> I have fixed the unbalanced ores. The torch seems to be left from my building sessions, I will remove that. I think I should better make the ravine with stone and not kill nodes. ~I am trying capturing to be mid hard.nut it doesn't matter to me if it is hard. :P.~ I will also look into making a better hint. Thanks for the feedback 👍

If you're talking about the torch in the pictures, I placed it there. The picture is just to show how the kill nodes look invisible when you look straight at them.

**Comment by src4026** [comment]
> If you're talking about the torch in the picture, I placed it there. The picture is just to show how the kill nodes look invisible when you look straight at them.

Oh! Ok :sweat_smile: . I had been putting a few torches during my builds so I got confused.

**Comment by src4026** [comment]
My reasons for the "et al"  are subjective. :P

**Comment by src4026** [comment]
I have removed the `killnodes` to make the cavern a combatable place. To prevent cases of tunnelling, I have added an unbreakable wall on both sides of the cavern. I have also changed the map treasure hint.

**Comment by a-blob** [comment]
Nice map. Everything seems to work.

File: cosmic_catastrophe/map.conf [hunk 1]
```diff
@@ -11,7 +11,7 @@ name = Cosmic Catastrophe
 phys_gravity = 1
 initial_stuff = return {"default:pick_steel","default:cobble 30","default:torch 5","ctf_ranged:pistol_loaded"}
 author = s20
-hint = The Reds shall forage the CT Fighter-20 and the Blues must probe the water springs to find the treasures.
+hint = The Reds shall forage the CT Fighter-20 andReds must conquer the flying fighter and the Blues must forage their Tower for the hidden treasures they seek.
```
Comment (a-blob) [inline cosmic_catastrophe/map.conf:14]: I'm not sure why the "andReds" doesn't have a space in between them.

Comment (src4026) [reply cosmic_catastrophe/map.conf:14]: Ah seems to be a typo =P

Comment (src4026) [reply cosmic_catastrophe/map.conf]: > I'm not sure why the "andReds" doesn't have a space in between them.

Fixed, I think

**Review by MoreKat — COMMENTED** [review]
I think you did a good job with the ravine. However, the positions of the ores for each team are now even more unbalanced. Red team has to climb/build up to a flying aircraft while all blue has to do is go down using a ladder to get to their treasure. 
As far as esthetics go, I will be honest, I find the map pretty ugly. I think it's because of the weird and inconsistent choice of materials that don't really go well together, like using snow, wool or steel/gold/bronze blocks for buildings. This is no reason for me to disapprove of the map though, I'm sure there are plenty of people who like the looks of it.

**Comment by src4026** [comment]
I will try to improve it as I can. :+1:

**Comment by src4026** [comment]
By the way @MoreKat , the ore rooms were always the same as they were. There are three ore rooms, with one with the indestructible blocks and a sign, which I've kept for a reason. It was always the case.

**Comment by src4026** [comment]
I have done the changes as in the comment of the previous push. I have not yet decided the material using which I will be making the wall. So that's still in discussion.

**Comment by src4026** [comment]
> I have done the changes as in the comment of the previous push. I have not yet decided the material using which I will be making the wall. So that's still in discussion.

The wall material has been set in the previous push to this PR.

**Comment by src4026** [comment]
I've planned something. So marked this as a draft for the time being. :heavy_check_mark:

**Comment by src4026** [comment]
Added a new sky box. It isn't fully done yet, the corners need to be blended properly. Ready for review.

**Comment by HobbitPower** [comment]
> Also, I will be adding stuff to the descriptions and be continually updating from time to time with tasks etc. 👍

Should I label it as WIP then? Sounds like it's not completely finished for a final review?

**Comment by src4026** [comment]
> Should I label it as WIP then? Sounds like it's not completely finished for a final review?

Yeah sure :+1:

**Comment by pl608** [comment]
Just a thought... could you update the screenshot to have your skybox?

**Comment by src4026** [comment]
I'll update it by Sunday along with making it ready for review :+1:

**Comment by olliy1or** [comment]
removing review required label until the map is finished

**Comment by src4026** [comment]
I'll close the PR for now. I don't have plans to work on this until late 2024 at the least

**Pull request closed without merging**